src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -912,8 +912,7 @@ public ListDatabasesResponse ListDatabases(ListDatabasesParams listDatabasesPara
             }
             ConnectionDetails connectionDetails = info.ConnectionDetails.Clone();
 
-            // Connect to master and query sys.databases
-            connectionDetails.DatabaseName = "master";
+            // Connect to database and query sys.databases

```

Comment:
This is necessary for Azure connections - querying sys.databases from a user DB on Azure only returns master and that user DB

Reply:
That being said - if the master connection fails falling back to querying the user DB doesn't seem like a terrible option

Reply:
Hmm, I guess we need a solution for when the user doesn't have access to master but has access to read databases.

Reply:
That comment came in later haha. Yeah that sounds good! I'll look into doing something there.